File: A-SOURCE_CODE/A-SERVICE/Shared/Models/TextSearch.cs
using Shared.Enumerations;

namespace Shared.Models
{
    public class TextSearch
    {
        /// <summary>
        ///     Value of text.
        /// </summary>
        public string Value { get; set; }

        /// <summary>
        ///     Mode of text filter
        /// </summary>
        public TextComparision Mode { get; set; }
    }
}

File: A-SOURCE_CODE/B-CLIENT/A-WEB/Administration/src/models/text-search.ts
import {TextSearchMode} from "../enumerations/text-search-mode";

/*
* Structure of text searching parameter.
* */
export class TextSearch{

    // String which is used for searching.
    public value: string;

    // Text searching mode.
    public mode: TextSearchMode;

    // Initiate text searching parameter.
    public constructor(){
        this.value = null;
        this.mode = TextSearchMode.contains;
    }
}
